src/hypertable_cache.c

```diff
@@ -158,6 +154,32 @@ hypertable_cache_get_entry_rv(Cache *cache, RangeVar *rv)
 	return hypertable_cache_get_entry(cache, RangeVarGetRelid(rv, NoLock, true));
 }
 
+Hypertable *
+hypertable_cache_get_entry_by_id(Cache *cache, int32 hypertable_id)
```

Comment:
Since the cache is on OID, it is impossible to do a lookup before scanning the `hypertable` table and retrieving the table and schema name that gives you the OID. I guess, perhaps, that a better way is to not create a full hypertable but to only scan for the matching tuple to get the OID, then do a regular cache lookup.

We could do a cache for hypertable_id->relid, but not sure the added complexity is worth it ower an index scan to find the hypertable tuple.

Reply:
Yeah I also think a new hypertable_id->relid cache is not necessary. But I don't see why you need to use the cache at all here. I like `to not create a full hypertable but to only scan for the matching tuple to get the OID`

Reply:
Did you look at my changes. This is exactly what it is doing now, as I suggested. It is scanning for the tuple to get the relid then it does a lookup in the regular cache.

Reply:
Just did. This LGTM now.